tcmur_cmd_handler.c

```diff
@@ -469,6 +474,216 @@ static int handle_read(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 	return async_handle_cmd(dev, cmd, read_work_fn);
 }
 
+struct unmap_state {
+	uint16_t bddl;
+	uint8_t *par;
+};
+
+static int unmap_work_fn(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
+{
+	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
+	struct unmap_state *unmap = cmd->cmdstate;
+	uint32_t block_size = tcmu_get_dev_block_size(dev);
+	uint64_t end_lba = tcmu_get_dev_num_lbas(dev) - 1;
+	struct iovec iovec;
+	size_t iov_cnt;
+	uint8_t *sense = cmd->sense_buf;
+	uint8_t *par;
+	uint16_t bddl = unmap->bddl, offset = 0;
+	void *iov_base;
+	int ret, i = 0;
+
+	iov_base = calloc(1, block_size);
```

Comment:
Will be fixed.

 I just test  it by using sg3 utils, and I haven't any ESXI environmemt at home, will do a verify in ESXI environmemt later.